authorship/data.py

```diff
@@ -40,43 +41,34 @@ def get(self):
         wordlist = [word for word, freq in get_frequent_words(speakers_wordlist, self._n_freqwords)]
 
         # build a dictionary of feature vectors
-        speakers = filter_texts_size(speakers_wordlist, wordlist, self._tokens_per_sample)
+        speakers_conv = filter_texts(speakers_wordlist, wordlist, self._min_num_of_words)
 
-        # convert to X, y
-        X, y = to_vector_size(speakers)
+        # convert to X, y (= spk id) but keep the conversation id
+        X, y, conv_ids = to_vector_size(speakers_conv)
 
-        return X, y
+        return X, y, conv_ids
 
     def __repr__(self):
-        return f'data(freqwords={self._n_freqwords}; ntokens={self._tokens_per_sample})'
+        return f'data(freqwords={self._n_freqwords})'
 
 
-class ExtractWords:
-    def __call__(self, texts):
-        yield from texts
-
-
-class RearrangeSamples:
-    def __init__(self, n, wordlist):
-        self.n = n
+class CreateFeatureVector:
+    def __init__(self, wordlist):
         self.wordlist = wordlist
 
-    def __call__(self, items):
-        for i in range(len(items) // self.n):
-            yield [word for word in items[i*self.n: (i+1)*self.n] if word in self.wordlist]
-
-
-class CreateFeatureVector:
-    def __init__(self, vocabulary):
-        self.vocabulary = vocabulary
+    def __call__(self, texts):
```

Comment:
what is `texts`? a list of lists of words? (missing documentation)

Reply:
documented